File: go/funciones.go
package main

import "fmt"

// Funciones
// Mostrar texto
func mostrarTexto() {
	fmt.Println("hola")
}

// Función multiplicar con valores internos
func multiplicar() {
	a := 5
	b := 8
	fmt.Println(a * b)
}

// Función multiplicar que usa variables globales
var a = 5
var b = 8

func multiplicarGlobal() {
	fmt.Println(a * b)
}

// Función multiplicar que devuelve un valor
func multiplicarReturn() int {
	a := 5
	b := 8
	return a * b
}

// Función para validar un dato
func validarDato() bool {
	if a == 5 {
		return true
	}
	return false
}

func main() {
	// Parte 1: Trabajar con listas (slice en Go)
	a := []int{1, 2, 3, 4, 5}
	b := []int{6, 7, 8, 9, 10}
	var c []int

	for i := 0; i < len(a); i++ {
		c = append(c, a[i]*b[i])
	}
	fmt.Println(c) // Imprime el resultado de multiplicar los elementos

	// Llamar a las funciones
	mostrarTexto()

	multiplicar()

	multiplicarGlobal()

	// Función que devuelve un valor
	resultado := multiplicarReturn()
	fmt.Println(resultado + 5)

	// Validación de un dato
	dato := validarDato()
	fmt.Println(dato)
}

File: java/funciones.java
import java.util.ArrayList;
import java.util.List;

public class Main {
    public static void main(String[] args) {
        // Inicia de los arreglos a y b
        int[] a = {1, 2, 3, 4, 5};
        int[] b = {6, 7, 8, 9, 10};
        
        // Crear una lista para almacenar los resultados
        //Integer envuelve un valor de tipo int en un objeto, lo que permite almacenar este valor en colecciones como listas//
        List<Integer> c = new ArrayList<>();
        
        // Bucle for para multiplicar los elementos correspondientes de los arreglos a y b
        for (int i = 0; i < a.length; i++) {
            c.add(a[i] * b[i]);
        }
        
        // Imprimir el resultado
        System.out.println(c);
        
        // Llamada a la función mostrarTexto
        mostrarTexto();
        
        // Llamada a la función multiplicar sin parámetros
        multiplicar();
        
        // Llamada a la función multiplicar con variables globales
        int aValue = 5;
        int bValue = 8;
        multiplicar(aValue, bValue);
        
        // Llamada a la función multiplicar con retorno y operación adicional
        int resultado = multiplicar();
        System.out.println(resultado + 5);
        
        // Llamada a la función validarDato
        boolean datoValido = validarDato(aValue);
        System.out.println(datoValido);
    }

    // Función para mostrar un texto
    public static void mostrarTexto() {
        System.out.println("hola");
    }

    // Función multiplicar sin parámetros (solo imprime)
    public static void multiplicar() {
        int a = 5;
        int b = 8;
        System.out.println(a * b);
    }

    // Función multiplicar con parámetros
    public static void multiplicar(int a, int b) {
        System.out.println(a * b);
    }

    // Función multiplicar con retorno
    public static int multiplicar() {
        int a = 5;
        int b = 8;
        return a * b;
    }

    // Función para validar un dato
    public static boolean validarDato(int a) {
        if (a == 5) {
            return true;
        } else {
            return false;
        }
    }
}
